packages/core/src/api/resolvers/entity/order-line-entity.resolver.ts

```diff
@@ -31,7 +33,19 @@ export class OrderLineEntityResolver {
         @Ctx() ctx: RequestContext,
         @Parent() orderLine: OrderLine,
     ): Promise<Asset | undefined> {
-        if (orderLine.featuredAsset !== undefined) {
+        if (!!orderLine.featuredAsset) {
```

Comment:
I thought more about this and I think it can be simplified even further to:

```ts
if (!orderLine.featuredAsset?.preview) {
  return orderLine.featuredAsset;
} else {
  return this.assetService.getFeaturedAsset(ctx, orderLine);
}
```

reasoning:
1. less code
2. fewer logical paths == easier to reason about
3. if any future refactors occur to how we fetch a featured asset, they will automatically apply to this code path == easier maintenance.

The comment is still worth having though to explain why we explicitly check for the presence of the `preview` property.

Reply:
Yeah, that makes sense; I was stuck on "change the least you need to" thinking that if featuredAsset is not truthy then we should avoid calling getFeaturedAsset, but I trust that you know better than I if there are any legitimate cases where that would happen, so I'll make and push the change.

Reply:
unless I'm reading wrong, though, you have your logic backwards, and it should be:

```typescript:
if (!orderLine.featuredAsset?.preview) {
    return this.assetService.getFeaturedAsset(ctx, orderLine);
} else {
    return orderLine.featuredAsset;
}
```

Reply:
yes you're right!